### PR TITLE
request to merge with further corrections to swagger YAML and JSON files for API specification

### PR DESCRIPTION
**Pull request by imcoleman**

### CODE REVIEW

File: fall-2017/icoleman/swagger/swagger.yaml [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+      parameters:
+        - name: name
+          in: query
+          description: name of user to search
+          required: false
+          type: string
+      responses:
+        '200':
+          description: successful operation
+          schema:
```
Comment (toddysm) [inline fall-2017/icoleman/swagger/swagger.yaml]: This should be array of Users

File: fall-2017/icoleman/swagger/swagger.yaml [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+            $ref: '#/definitions/User'
+        - name: userId
+          in: path
+          description: User id to add
+          required: true
+          type: integer
+          format: int64
+      responses:
+        '400':
```
Comment (toddysm) [inline fall-2017/icoleman/swagger/swagger.yaml]: There is also 200 respose

File: fall-2017/icoleman/swagger/swagger.yaml [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+        - application/json
+      parameters:
+        - in: body
+          name: body
+          description: User object that needs to be added
+          required: true
+          schema:
+            $ref: '#/definitions/User'
+      responses:
+        '405':
```
Comment (toddysm) [inline fall-2017/icoleman/swagger/swagger.yaml]: There is also 200 response

File: fall-2017/icoleman/swagger/swagger.yaml [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+        - application/json
+      parameters:
+        - name: userId
+          in: path
+          description: User id to delete
+          required: true
+          type: integer
+          format: int64
+      responses:
+        '400':
```
Comment (toddysm) [inline fall-2017/icoleman/swagger/swagger.yaml]: There is also 200 response

File: fall-2017/icoleman/swagger/swagger.yaml [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+        - name: title
+          in: query
+          description: title of task to search
+          required: false
+          type: string
+      responses:
+        '200':
+          description: successful operation
+          schema:
+            $ref: '#/definitions/Task'
```
Comment (toddysm) [inline fall-2017/icoleman/swagger/swagger.yaml]: It should be array of tasks

File: fall-2017/icoleman/swagger/swagger.yaml [hunk 6]
```diff
@@ -0,0 +1,340 @@
...
+        - name: userId
+          in: path
+          description: User id to delete
+          required: true
+          type: integer
+          format: int64
+      responses:
+        '405':
```
Comment (toddysm) [inline fall-2017/icoleman/swagger/swagger.yaml]: There should be 200 response

File: fall-2017/icoleman/swagger/swagger.yaml [hunk 7]
```diff
@@ -0,0 +1,340 @@
...
+          type: integer
+          format: int64
+        - name: taskId
+          in: path
+          description: ID of task to return
+          required: true
+          type: integer
+          format: int64
+      responses:
+        '400':
```
Comment (toddysm) [inline fall-2017/icoleman/swagger/swagger.yaml]: There should be 200 response

File: fall-2017/icoleman/swagger/swagger.yaml [hunk 8]
```diff
@@ -0,0 +1,340 @@
...
+          type: integer
+          format: int64
+        - name: taskId
+          in: path
+          description: Task id to delete
+          required: true
+          type: integer
+          format: int64
+      responses:
+        '400':
```
Comment (toddysm) [inline fall-2017/icoleman/swagger/swagger.yaml]: There should be 200 response

**Comment by imcoleman** [comment]
Hi Toddy,
I realize reading your current comments that I have not understood your
previous comments, because now I see that I may have corrected something
which didn't need to be corrected.
Previously, you said that /users "This actually returns array of users", by
which I thought you wanted it to be edited to return a user which has been
searched for.  But now your comment says /users "this should be array of
Users".  Was it correct before, and if so, what needed to be corrected on
that item?
Similarly, you said that /users/{userId}/tasks "This is also array of
tasks" -- it was already an array of tasks, but I thought you wanted me to
add additional functionality to search for a single task, but now your
comment says "It should be array of tasks".
Thanks,
Ilsa

**Pull request closed without merging**